File: src/client/java/vergisst/minecraftmod/weaponthrow/client/packet/SPacketThrow.java
package vergisst.minecraftmod.weaponthrow.client.packet;

import net.fabricmc.fabric.api.client.networking.v1.ClientPlayNetworking;
import vergisst.minecraftmod.weaponthrow.client.handler.EventsHandler;
import vergisst.minecraftmod.weaponthrow.client.handler.PacketHandler;
import vergisst.minecraftmod.weaponthrow.packets.BasePacket;
import vergisst.minecraftmod.weaponthrow.packets.PacketIdentifiers;

import java.util.UUID;

@Deprecated
public class SPacketThrow extends BasePacket {

    public SPacketThrow(UUID uuid, int maxChargeTime, boolean isCharging) {
        super(PacketIdentifiers.SPACKET_THROW);
        buf.writeUuid(uuid);
        buf.writeVarInt(maxChargeTime);
        buf.writeBoolean(isCharging);
    }

    public static void register() {

        ClientPlayNetworking.registerGlobalReceiver(PacketIdentifiers.SPACKET_THROW, (client, handler, buf, responseSender) -> {
            UUID uuid = buf.readUuid();
            int maxChargeTime = buf.readVarInt();
            boolean isCharging = buf.readBoolean();

            client.execute(() -> {
                EventsHandler.onSeverUpdate(uuid, maxChargeTime, isCharging);
            });
        });

    }

}

File: src/client/kotlin/vergisst/minecraftmod/weaponthrowlite/client/packets/SPacketThrow.kt
package vergisst.minecraftmod.weaponthrowlite.client.packets

import net.fabricmc.fabric.api.client.networking.v1.ClientPlayNetworking
import net.fabricmc.fabric.api.networking.v1.PacketSender
import net.minecraft.client.MinecraftClient
import net.minecraft.client.network.ClientPlayNetworkHandler
import net.minecraft.network.PacketByteBuf
import vergisst.minecraftmod.weaponthrowlite.client.handlers.EventsHandler
import vergisst.minecraftmod.weaponthrowlite.handlers.PacketsIdentifier
import vergisst.minecraftmod.weaponthrowlite.packets.BasePacket
import java.util.*

//class SPacketThrow: BasePacket(PacketsIdentifier.SPACKET_THROW) {
//
//}

class SPacketThrow(uuid: UUID?, maxChargeTime: Int, isCharging: Boolean) : BasePacket(PacketsIdentifier.SPACKET_THROW) {
    init {
        buf?.writeUuid(uuid)
        buf?.writeVarInt(maxChargeTime)
        buf?.writeBoolean(isCharging)
    }

    companion object {
        fun register() {
            ClientPlayNetworking.registerGlobalReceiver(
                PacketsIdentifier.SPACKET_THROW,
                ClientPlayNetworking.PlayChannelHandler { client: MinecraftClient?, handler: ClientPlayNetworkHandler?, buf: PacketByteBuf?, responseSender: PacketSender? ->
                    val uuid = buf!!.readUuid()
                    val maxChargeTime = buf.readVarInt()
                    val isCharging = buf.readBoolean()
                    client!!.execute(Runnable {
                        EventsHandler.onServerUpdate(uuid, maxChargeTime, isCharging)
                    })
                })
        }
    }
}
